apps/io-func-wallet-solution-backend/src/infra/pdv-tokenizer/client.ts

```diff
@@ -0,0 +1,121 @@
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/function";
+import * as t from "io-ts";
+import { v4 as uuidv4 } from "uuid";
+import { PdvTokenizerHealthCheck } from "./health-check";
+import { UserIdRepository } from "@/user";
+import { PdvTokenizerApiClientConfig } from "@/app/config";
+
+const Token = t.type({
+  token: NonEmptyString,
+});
+
+const Pii = t.type({
+  pii: FiscalCode,
+});
+
+export class PdvTokenizerClient
+  implements UserIdRepository, PdvTokenizerHealthCheck
+{
+  #baseURL: string;
+  #options: RequestInit;
+
+  constructor({ baseURL, apiKey }: PdvTokenizerApiClientConfig) {
+    this.#baseURL = baseURL;
+    this.#options = {
+      headers: {
+        "x-api-key": apiKey,
+        Accept: "application/json, text/plain",
+        "Content-Type": "application/json",
+      },
+    };
+  }
+
+  getUserIdByFiscalCode = (fiscalCode: FiscalCode) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          const result = await fetch(
+            `${this.#baseURL}/tokenizer/v1/tokens/search`,
```

Comment:
what do you think about using the `URL()` constructor here?
ex: `new URL("/tokenizer/v1/tokens/search", this.#baseURL);`

Reply:
done in 68f2f58